### PR TITLE
[Token] create a general token event store for new token events 

### PR DESCRIPTION
**Pull request by areshand**
### Description
1. create a new token event store for all future new token events
2. reorganize the code based on entry, public and private/friend to different sections for easy navigating
3. use transfer event as an example for this general token store. 

### Test Plan
add unit test

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/4401)

### CODE REVIEW

File: aptos-move/framework/aptos-token/sources/token.move [hunk 1]
```diff
@@ -376,6 +377,145 @@ module aptos_token::token {
         *opt_in_flag = opt_in;
     }
 
+    /// mutate the token property and save the new property in TokenStore
+    /// if the token property_version is 0, we will create a new property_version per token to generate a new token_id per token
+    /// if the token property_version is not 0, we will just update the propertyMap and use the existing token_id (property_version)
+    public entry fun mutate_token_properties(
```
Comment (areshand) [inline aptos-move/framework/aptos-token/sources/token.move:383]: this is simply a copy and re-organize

File: aptos-move/framework/aptos-token/sources/token.move [hunk 2]
```diff
@@ -376,6 +377,145 @@ module aptos_token::token {
...
+        let i = 0;
+        // give a new property_version for each token
+        while (i < amount) {
+            mutate_one_token(account, token_owner, token_id, keys, values, types);
+            i = i + 1;
+        };
+    }
+
+    /// create token with raw inputs
+    public entry fun create_token_script(
```
Comment (areshand) [inline aptos-move/framework/aptos-token/sources/token.move:416]: this is simply a copy and re-organize

File: aptos-move/framework/aptos-token/sources/token.move [hunk 3]
```diff
@@ -376,6 +377,145 @@ module aptos_token::token {
...
+        );
+
+        mint_token(
+            account,
+            tokendata_id,
+            balance,
+        );
+    }
+
+    public entry fun burn(
```
Comment (areshand) [inline aptos-move/framework/aptos-token/sources/token.move:457]: this is simply a copy and re-organize

File: aptos-move/framework/aptos-token/sources/token.move [hunk 4]
```diff
@@ -1170,6 +1117,81 @@ module aptos_token::token {
         }
     }
 
+    //
+    // Private or friend-only functions
+    //
+
+    fun update_token_property_internal(
```
Comment (areshand) [inline aptos-move/framework/aptos-token/sources/token.move:1124]: this is simply a copy and re-organize as well

**Review by movekevin — APPROVED** [review]

**Comment by bowenyang007** [comment]
Can we see this in action? If we can get a transaction in testnet i can try to parse it out. 

One concern i have is that property map will be bcs encoded which means that the node API will be effectively unusable for those events. It'll be gibberish and people won't know what to do with them. cc @gregnazario

**Review by bowenyang007 — CHANGES_REQUESTED** [review]
Wanted to make sure that we don't land this before having proper testing...unit tests aren't good enough to test if a standard's usability. Let's do better here especially given this is designed to be the standard for all events moving forward.

**Review by davidiw — CHANGES_REQUESTED** [review]
Thanks for pushing on this... after discussion today, the indexer folks do not feel the need for this as a mainnet requirement. the reality is that due to the lack of ecosystem projects using many of these interfaces, we should spend more time to evaluate the right direction here.

I'd also like for us to consider how we might make an event stream that doesn't require serialization

**Pull request closed without merging**